### PR TITLE
Adds aws_iam_saml_provider

### PR DESCRIPTION
**Pull request by tmonk42**
### Description

Add a new resource and tests for IAM SAML Provider.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [X] New functionality includes integration tests/controls
- [N/A] New Terraform resources
- [X] Documentation provided or updated for resources 
- [N/A] All Integration Tests pass
- [X] All Unit Tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_saml_provider.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    super(opts)
+    validate_parameters([:saml_provider_arn])
+
+    catch_aws_errors do
+      @provider = @aws.iam_client.get_saml_provider(saml_provider_arn: opts[:saml_provider_arn])
+      @arn = opts[:saml_provider_arn]
+      @create_date = @provider.create_date
+      @valid_until = @provider.valid_until
+      # Not currently doing anything with metadata doc, future improvement?
+      # @saml_metadata_document = @provider.saml_metadata_document
```
Comment (r-fennell) [inline libraries/aws_iam_saml_provider.rb]: I would suggest simply exposing `saml_metadata_document` now - save us revisiting in the future. Once it's exposed we are then exposing everything returned from the API and can therefor make `@provider` a local variable and remove it as an `attr_reader`.

File: libraries/aws_iam_saml_provider.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    end
+  end
+
+  def exists?
+    !@provider.saml_metadata_document.nil?
```
Comment (r-fennell) [inline libraries/aws_iam_saml_provider.rb]: Need `saml_metadata_document` populated for this to work.

File: libraries/aws_iam_saml_providers.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters([])
+    @table = fetch_data
+  end
+
+  def fetch_data
+    provider_rows = []
+    catch_aws_errors do
+      @providers = @aws.iam_client.list_saml_providers
```
Comment (r-fennell) [inline libraries/aws_iam_saml_providers.rb]: `@providers` should really be a local variable here

File: test/unit/resources/aws_iam_saml_provider_test.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
+require 'aws-sdk-core'
+require 'aws_iam_saml_provider'
+require 'helper'
+require_relative 'mock/iam/aws_iam_saml_provider_mock'
+
+class AwsIamSamlProviderTest < Minitest::Test
```
Comment (r-fennell) [inline test/unit/resources/aws_iam_saml_provider_test.rb]: 👍

File: test/unit/resources/mock/iam/aws_iam_saml_provider_mock.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+require_relative '../aws_base_resource_mock'
+
+class AwsIamSamlProviderMock < AwsBaseResourceMock
+
+  # Default attributes.
+  def initialize
+    super
+    @provider = OpenStruct.new(
```
Comment (r-fennell) [inline test/unit/resources/mock/iam/aws_iam_saml_provider_mock.rb]: I like this as long as it's what we expect from the API.

Comment (tmonk42) [reply test/unit/resources/mock/iam/aws_iam_saml_provider_mock.rb]: Yup, API provides a struct

**Comment by skpaterson** [comment]
Had a look at whether it would be possible to create a dummy SAML provider for integration tests but this was non-trivial.

**Pull request merged**